### PR TITLE
Enhance E2E job

### LINKED ISSUE #623: Integrate E2E with Boskos and ibmcloud janitor
/kind feature
/area provider/ibmcloud

**Describe the solution you'd like**
Currently the periodic E2E job is configured to use the existing Boksos janitor. Make changes accordingly to integrate it with ibmcloud janitor.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Amulyam24**
**What this PR does / why we need it**:
This PR adds changes to use Boskos resources of type powervs-service instead of using network resource type.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #623 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Enhance E2E job to use powervs-service resources
```

### CODE REVIEW

File: scripts/ci-e2e.sh [hunk 1]
```diff
@@ -126,7 +123,7 @@ main(){
         HEART_BEAT_PID=$(echo $!)
     fi
 
-    if [[ "${E2E_FLAVOR}" == "powervs" ]]; then
+    if [[ "${E2E_FLAVOR}" == "powervs" || "${E2E_FLAVOR}" == "md-remediation" ]]; then
```
Comment (mkumatag) [inline scripts/ci-e2e.sh:126]: `md-remediation` where is this flavour present?

File: scripts/ci-e2e.sh [hunk 2]
```diff
@@ -91,9 +87,10 @@ init_network_powervs(){
 
 prerequisites_powervs(){
     # Assigning PowerVS variables
-    export IBMPOWERVS_IMAGE_NAME=${IBMPOWERVS_IMAGE_NAME:-"capibm-powervs-centos-streams8-1-22-4"}
-    export IBMPOWERVS_SERVICE_INSTANCE_ID=${IBMPOWERVS_SERVICE_INSTANCE_ID:-"0f28d13a-6e33-4d86-b6d7-a9b46ff7659e"}
-    export IBMPOWERVS_NETWORK_NAME=${BOSKOS_RESOURCE_NAME:-"capi-e2e-test"}
+    export IBMPOWERVS_SSHKEY_NAME=${IBMPOWERVS_SSHKEY_NAME:-"powercloud-bot-key"}
+    export IBMPOWERVS_IMAGE_NAME=${IBMPOWERVS_IMAGE_NAME:-"capibm-powervs-centos-streams8-1-24-2"}
+    export IBMPOWERVS_SERVICE_INSTANCE_ID=${BOSKOS_RESOURCE_ID:-"0f28d13a-6e33-4d86-b6d7-a9b46ff7659e"}
+    export IBMPOWERVS_NETWORK_NAME="capi-net-$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | head --bytes 5)"
```
Comment (mkumatag) [inline scripts/ci-e2e.sh:93]: Going forward we need to override these values while calling these script

File: scripts/ci-e2e.sh [hunk 3]
```diff
@@ -37,14 +37,10 @@ ARCH=$(uname -m)
 PVSADM_VERSION=${PVSADM_VERSION:-"v0.1.4-alpha.3"}
 E2E_FLAVOR=${E2E_FLAVOR:-}
 REGION=${REGION:-"us-south"}
-RESOURCE_GROUP=${RESOURCE_GROUP:-"prow-resource-group"}
 
 trap cleanup EXIT
 
 cleanup(){
-    # Delete the created ports for the network instance
-    [ -n "${NEW_PORT}" ] && ./pvsadm delete port --network ${IBMPOWERVS_NETWORK_NAME} --port-id ${PORT_ID} --instance-id ${IBMPOWERVS_SERVICE_INSTANCE_ID}
```
Comment (mkumatag) [inline scripts/ci-e2e.sh:46]: why are we not deleting the port anymore?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: scripts/ci-e2e.sh (hunk 1 above)
Comment (Amulyam24) [reply scripts/ci-e2e.sh:126]: This flavour is present as a template here - https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/tree/main/test/e2e/data/templates

File: scripts/ci-e2e.sh (hunk 3 above)
Comment (Amulyam24) [reply scripts/ci-e2e.sh:46]: As the test uses resources of type powervs service instance, we are depending on the ibmcloud-janitor to take care of the cleanup of ports, network and instances.

File: scripts/ci-e2e.sh (hunk 1 above)
Comment (mkumatag) [reply scripts/ci-e2e.sh:126]: As I mentioned earlier for the long run we need to find a way to patch the existing template instead of having a new one.

**Review by mkumatag — APPROVED** [review]

File: scripts/ci-e2e.sh (hunk 1 above)
Comment (Amulyam24) [reply scripts/ci-e2e.sh:126]: Will be raising a follow-up PR for it.

**Pull request merged**